src/main/java/com/yahoo/sketches/theta/ConcurrentUpdateSketch.java

```diff
@@ -0,0 +1,83 @@
+package com.yahoo.sketches.theta;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * @author eshcar
+ */
+public class ConcurrentUpdateSketch extends UpdateSketchComposition {
+
+  private static int PARALLELISM_LEVEL = 3;
+  protected static ExecutorService PROPAGATION_EXECUTOR_SERVICE =
+      Executors.newWorkStealingPool(PARALLELISM_LEVEL);
+
+  private volatile long theta_;
+  private volatile double estimation_;
+  // A flag to coordinate between several propagation threads
+  private AtomicBoolean propagationInProgress_;
+
+  // package visibility constructors - to be created only by factory
+  ConcurrentUpdateSketch(final UpdateSketch delegattee) {
+    super(delegattee);
+    theta_ = getThetaLong();
+    estimation_ = 0;
+    propagationInProgress_ = new AtomicBoolean(false);
+  }
+
+  public void propagate(final Sketch sketchIn, final AtomicBoolean propagationInProgress) {
```

Comment:
Why does this need to be public?

Reply:
I'll check, perhaps package visibility is enough